src/services/accounts/AccountsBalanceInfoService.ts

```diff
@@ -161,14 +174,89 @@ export class AccountsBalanceInfoService extends AbstractService {
 				at,
 				nonce,
 				tokenSymbol: token,
-				free,
-				reserved,
-				miscFrozen,
-				feeFrozen,
-				locks,
+				free: this.inDenominationBal(denominate, free, decimal),
+				reserved: this.inDenominationBal(denominate, reserved, decimal),
+				miscFrozen: this.inDenominationBal(denominate, miscFrozen, decimal),
+				feeFrozen: this.inDenominationBal(denominate, feeFrozen, decimal),
+				locks: this.inDenominationLocks(denominate, locks, decimal),
 			};
 		} else {
 			throw new BadRequest('Account not found');
 		}
 	}
+
+	/**
+	 * Apply a denomination to a balance depending on the chains decimal value.
+	 *
+	 * @param balance free balance available encoded as Balance
+	 * @param dec The chains given decimal value
+	 */
+	private applyDenominationBalance(balance: Balance, dec: number): string {
+		const strBalance = balance.toString();
+
+		// We dont want to denominate a zero balance or zero decimal
+		if (strBalance === '0' || dec === 0) {
+			return strBalance;
```

Comment:
hmm,

this means that `0`will be => `0` and not `0.00` correct?
perhaps it would be better to add the decimals for zero to for consistency